core/base-service/base-json.js

```diff
@@ -34,10 +34,22 @@ class BaseJsonService extends BaseService {
    *    and custom error messages e.g: `{ 404: 'package not found' }`.
    *    This can be used to extend or override the
    *    [default](https://github.com/badges/shields/blob/master/core/base-service/check-error-response.js#L5)
+   * @param {object} [attrs.customExceptions={}] Key-value map of got network exception codes
+   *    and an object of params to pass when we construct an Inaccessible exception object
+   *    e.g: `{ ECONNRESET: { prettyMessage: 'connection reset' } }`.
+   *    See {@link https://github.com/sindresorhus/got/blob/main/documentation/7-retry.md#errorcodes got error codes}
+   *    for allowed keys
+   *    and {@link module:core/base-service/errors~RuntimeErrorProps} for allowed values
    * @returns {object} Parsed response
    * @see https://github.com/sindresorhus/got/blob/main/documentation/2-options.md
    */
-  async _requestJson({ schema, url, options = {}, errorMessages = {} }) {
+  async _requestJson({
+    schema,
+    url,
+    options = {},
+    errorMessages = {},
+    customExceptions = {},
```

Comment:
`errorMessages` and `customExceptions` do somewhat similar but importantly different things. Naming things is hard.

Reply:
ack yeah I see that being tricky down the road since both of these are (ultimately) used in the setting of error messages.

I feel like in a perfect world (one where no refactoring would be required), I'd be inclined to keep a single top level object for errors and maybe have it have two properties, one that would point to an object that's our standard http response status code to error message dictionary we have today, and the other would be the new I feel like a single top level object that had two properties, one being new dictionary that maps nodejs error codes to a desired error message.

Thinking on a bit more, I suppose having those two mappings separate as you do here is fine; I think it's just the first having squatted on the highly generic `errorMessages` name that feels problematic.

What if we come up with a more descriptive name for this one for now, e.g. (overly verbosely) `nodeErrorCodeToShieldsExceptions` and then we could potentially consider renaming `errorMessages` to something similarly more descriptive down the road?

Reply:
Yeah broadly agreed. Just thinking about the ergonomics of consuming this at the service layer...

Mapping HTTP error codes to error messages is really common - we do stuff like

```js
errorMessages: { 404: 'package not found' }
```

all over the place, so lets keep that easy to do in a convenient way.

I've designed this feature (provisionally lets call it `customExceptions`) to be relatively generic, but I'd expect us to actually use this quite rarely.

I think given that, I'd prefer to keep a flatter structure and avoid making `errorMessages` a nested object. In the vast majority of cases we would only care about one of the keys. If we can come up with 2 good names we would need to change the code in loads of places anyway, so lets make those two good names the top-level names rather than introduce another layer of nesting.

i.e: If we need to go and change a load of existing service classes, I'd rather do a big find and replace changing

```js
errorMessages: { 404: 'package not found' }
```

to

```js
httpErrors: { 404: 'package not found' }
```

than

```js
errorMessages: { httpErrors: { 404: 'package not found' } }
```

or whatever.

If we ignore the fact that there is loads of existing code using `errorMessages` and we were naming these two concepts from scratch today, what do you think would be the ideal names/API for these things? I guess the things we currently call `errorMessages` are really HTTP Status Codes. The thing I've called `customExceptions` are really System Error Codes. How about `httpErrors` and `systemErrors` ?

Reply:
> How about httpErrors and systemErrors ?

Yup, love it. Definitely clears up the naming overlap and is much more succinct than the ones I threw out there